### ISSUE TITLE
model.fit failing with segfault in libdrm_amdgpu.so.1.0.0 then in libamdocl64.so

### ISSUE THREAD

**Issue by rlong011** [plaidml/plaidml#189]
os: ubuntu 16.04.3
     Number of platforms                               1
     Platform Name                                   AMD Accelerated Parallel Processing
     Platform Vendor                                 Advanced Micro Devices, Inc.
    Platform Version                                OpenCL 2.0 AMD-APP (2482.3)
    Platform Profile                                FULL_PROFILE
    Platform Extensions                             cl_khr_icd cl_amd_event_callback cl_amd_offline_devices 
    Platform Extensions function suffix             AMD

  Platform Name                                   AMD Accelerated Parallel Processing
Number of devices                                 5
  Device Name                                     Ellesmere
  Device Vendor                                   Advanced Micro Devices, Inc.
  Device Vendor ID                                0x1002
  Device Version                                  OpenCL 1.2 AMD-APP (2482.3)
  Driver Version                                  2482.3
  Device OpenCL C Version                         OpenCL C 1.2 
  Device Type                                     GPU
  Device Profile                                  FULL_PROFILE
  Device Board Name (AMD)                         AMD Radeon (TM) RX 480 Graphics
  Device Topology (AMD)                           PCI-E, 01:00.0
  Max compute units                               36
  SIMD per compute unit (AMD)                     4
  SIMD width (AMD)                                16
  SIMD instruction width (AMD)                    1
  Max clock frequency                             1266MHz
  Graphics IP (AMD)                               8.0
  Device Partition                                (core)
    Max number of sub-devices                     36
    Supported partition types                     none specified
  Max work item dimensions                        3
  Max work item sizes                             256x256x256
  Max work group size                             256
  Preferred work group size multiple              64
  Wavefront width (AMD)                           64
  Preferred / native vector si

Running 
       model = Sequential()
       model.add(Embedding(max_features, 128, input_shape = (maxlen,)))
       model.add(LSTM(64, dropout=0.2, recurrent_dropout=0.2))
       model.add(Dense(1, activation='sigmoid'))

        model.compile(loss='binary_crossentropy',
                  optimizer='adam',
                  metrics=['accuracy'])
                 INFO:plaidml:Opening device "opencl_amd_ellesmere.0"
        model.fit(X_train_pad, y_train,
          batch_size=32,
          epochs=2,
          validation_split=0.3)
       
          Train on 17500 samples, validate on 7500 samples
           Epoch 1/2

           INFO:plaidml:Analyzing Ops: 1403 of 11727 operations complete
           INFO:plaidml:Analyzing Ops: 3824 of 11727 operations complete
           INFO:plaidml:Analyzing Ops: 6882 of 11727 operations complete
           INFO:plaidml:Analyzing Ops: 9691 of 11727 operations complete
Ist run I get 
            segfault at 18 ip 00007f414baac835 sp 00007f410992ebe0 error 4 in                 libdrm_amdgpu.so.1.0.0[7f414baa8000+d000]

2nd try
            segfault at 18 ip 00007f218dd8130d sp 00007f2189b70790 error 4 in libamdocl64.so[7f218d6d8000+38ed000]

**Comment by brianretford** [comment]
Training embeddings is currently the main thing plaidml is terrible at, as our scatter kernel is just not very good. It still shouldn't segfault, but our current approach utilizes so much memory in some instances that I wouldn't be suprised.

Out of curiosity, which AMD drivers are you using? It's my impression that the newer ROCm stuff is much better.

**Comment by brianretford** [comment]
duplicate of #70
